ComponentKit/Core/Scope/CKComponentScopeHandle.mm

```diff
@@ -85,12 +85,14 @@ - (instancetype)initWithListener:(id<CKComponentStateListener>)listener
 }
 
 - (instancetype)newHandleWithStateUpdates:(const CKComponentStateUpdateMap &)stateUpdates
+                       componentScopeRoot:(CKComponentScopeRoot *)componentScopeRoot
 {
   id updatedState = _state;
   const auto range = stateUpdates.equal_range(_globalIdentifier);
   for (auto it = range.first; it != range.second; ++it) {
     updatedState = it->second(updatedState);
   }
+  [componentScopeRoot registerAnnounceableEventsForController:_controller];
```

Comment:
I explicitly decided to pass in the component scope root instead of grabbing it from the current scope. It felt a bit safer.